sequencer/src/state.rs

```diff
@@ -332,6 +332,12 @@ impl HotShotState<SeqTypes> for ValidatedState {
         // through returned value.
         let mut validated_state = self.clone();
 
+        // validate `ChainConfig`
+        if proposed_header.chain_config != instance.chain_config.into() {
```

Comment:
I think we should add a test that hits this. Or do we have it and I'm not seeing it?